packages/testing/src/execution_testing/client_clis/file_utils.py

```diff
@@ -22,7 +24,19 @@ def dump_files_to_directory(output_path: str, files: Dict[str, Any]) -> None:
             os.makedirs(os.path.join(output_path, rel_path), exist_ok=True)
         file_path = os.path.join(output_path, file_rel_path)
         with open(file_path, "w") as f:
-            if isinstance(file_contents, BaseModel) or isinstance(
+            if isinstance(file_contents, (LazyAllocStr, LazyAllocJson)):
+                if isinstance(file_contents, LazyAllocJson):
+                    dump(file_contents.raw, f, ensure_ascii=True, indent=4)
+                else:
+                    # Validate and serialize LazyAlloc
+                    f.write(
+                        file_contents.get().model_dump_json(
+                            indent=4,
+                            exclude_none=True,
+                            by_alias=True,
+                        )
+                    )
```

Comment:
```suggestion
                    f.write(file_contents.raw)
```
I think in this case we can write raw directly since it's text.

Reply:
@marioevz don't we want to validate here though? See my comment here https://github.com/ethereum/execution-specs/pull/1852#discussion_r2599479518